styx-scheduler-service/src/main/java/com/spotify/styx/StyxScheduler.java

```
@@ -570,11 +571,28 @@ private void setupMetrics(
           workflow.id(),
           () -> stateManager.getActiveStatesCount(workflow.id()));
 
+      final Optional<Workflow> existingWorkflow = cache.workflow(workflow.id());
+      if (existingWorkflow.isPresent()) {
+        if (!isGreaterOrEqualApiVersion(workflow, existingWorkflow.get())) {
+          return;
```

Silently returning with a 200 OK? I think that's lying.